packages/openstef-core/src/openstef_core/feature_engineering/validation_transforms/missing_values_transform.py

```diff
@@ -0,0 +1,209 @@
+
+    @model_validator(mode="after")
+    def validate_fill_value_with_strategy(self) -> "MissingValuesTransform":
+        """Validate that fill_value is provided when strategy is CONSTANT.
+
+        Returns:
+            The validated model instance.
+
+        Raises:
+            ValueError: If imputation_strategy is CONSTANT but fill_value is None.
+        """
+        if self.imputation_strategy == ImputationStrategy.CONSTANT and self.fill_value is None:
+            raise ValueError("fill_value must be provided when imputation_strategy is CONSTANT")
+        return self
+
+    def __init__(self, **data: Any):
+        """Initialize the MissingValuesTransform with the given configuration."""
+        super().__init__(**data)
+
+        self.imputer = SimpleImputer(
+            strategy=self.imputation_strategy.value,
+            fill_value=self.fill_value,
+            missing_values=self.missing_value,
+            keep_empty_features=False,
+        )
+        self.imputer.set_output(transform="pandas")
+
+    def _drop_empty_features(self, data: pd.DataFrame) -> pd.DataFrame:
+        """Drop features that contain only missing values.
+
+        Args:
+            data: The input DataFrame.
+
+        Returns:
+            A DataFrame with empty features removed.
+        """
+        all_missing_mask = data.isna().all() if np.isnan(self.missing_value) else (data == self.missing_value).all()
+
+        for col in data.columns[all_missing_mask]:
+            _logger.warning("Dropped column '%s' from dataset because it contains only missing values.", col)
+
+        non_empty_columns = data.columns[~all_missing_mask]
+        return data[non_empty_columns]
+
+    def _remove_trailing_null_rows(self, data: pd.DataFrame) -> pd.DataFrame:
+        """Remove trailing rows that contain null values in specified features.
+
+        This method removes rows from the end of the dataset where specified features
+        (defined in no_fill_future_values_features) contain null values. This
+        is useful for cleaning datasets where future values should not be filled and
+        trailing nulls indicate incomplete data.
+
+        Args:
+            data: The input pandas DataFrame containing the data
+                to be processed.
+
+        Returns:
+            A pandas DataFrame with trailing null rows removed for the specified
+                features. If no features are configured for checking or none of the
+                configured features exist in the data, returns the original data unchanged.
+        """
+        if not self.no_fill_future_values_features:
+            return data
+
+        features_to_check = [f for f in self.no_fill_future_values_features if f in data.columns]
+        for feature in self.no_fill_future_values_features:
+            if feature not in data.columns:
+                _logger.warning("Feature '%s' not found in dataset columns.", feature)
```

Comment:
I would avoid logging multiple warnings and instead combine them into one.
```suggestion
        missing_columns = set(self.no_fill_future_values_features) - set(data.columns)
        if missing_columns:
            _logger.warning("Features '%s' not found in dataset columns.", missing_columns)
```

Reply:
I wonder if we should make it strict instead. Instead of giving a warning throw an error, because the input is invalid. Also fewer edge cases to deal with in the transforms.